presto-raptor/src/test/java/com/facebook/presto/raptor/metadata/TestShardCleaner.java

```diff
@@ -189,6 +189,64 @@ public void testDeleteOldShards()
     }
 
     @Test
+    public void testCleanLocalShardsAtStartup()
+            throws Exception
+    {
+        assertEquals(cleaner.getLocalShardsCleaned().getTotalCount(), 0);
+        TestingShardDao shardDao = dbi.onDemand(TestingShardDao.class);
+        MetadataDao metadataDao = dbi.onDemand(MetadataDao.class);
+
+        long tableId = metadataDao.insertTable("test", "test", false, false, null, 0);
+
+        UUID shard1 = randomUUID();
+        UUID shard2 = randomUUID();
+        UUID shard3 = randomUUID();
+
+        Set<UUID> shards = ImmutableSet.of(shard1, shard2, shard3);
+
+        for (UUID shard : shards) {
+            shardDao.insertShard(shard, tableId, null, 0, 0, 0);
+            createShardFile(shard);
+            assertTrue(shardFileExists(shard));
+        }
+
+        int node1 = shardDao.insertNode("node1");
+        int node2 = shardDao.insertNode("node2");
+
+        // shard 1: referenced by this node
+        // shard 2: not referenced
+        // shard 3: referenced by other node
+
+        shardDao.insertShardNode(shard1, node1);
+        shardDao.insertShardNode(shard3, node2);
+
+        // advance time within clean time
+        Duration cleanTime = new ShardCleanerConfig().getLocalCleanTime();
+        ticker.increment(cleanTime.toMillis() - 1, MILLISECONDS);
+
+        // try non-startup mode. This should not clean any shards
```

Comment:
I see you point. However, the goal of this test is to verify expected behavior of the method. You don't need to prove the method's usefulness or compare it with other ones in this test.
I think you can create a setup with incremented ticker, but I don't think you should run another cleaning method inside this test.